backend/EasyMeets.Core/EasyMeets.Core.BLL/Services/AvailabilityService.cs

```diff
@@ -241,14 +241,26 @@ public async Task<AvailabilitySlotDto> GetAvailabilitySlotById(long id)
 
             if (!updateAvailabilityDto.Schedule.WithTeamMembers)
             {
-                var scheduleId = availabilitySlot.SlotMembers.First(member => member.MemberId == currentUser.Id).ScheduleId;
+                var scheduleId = availabilitySlot.SlotMembers.First(member => member.MemberId == currentUser.Id)
+                    .ScheduleId;
                 var updatedExclusionDateIds = updateAvailabilityDto.Schedule.ExclusionDates.Select(date => date.Id);
-                var deletedExclusionDates = await _context.ExclusionDates
-                    .Where(date => date.ScheduleId == scheduleId &&
-                                   updatedExclusionDateIds.All(updatedDateId => updatedDateId != date.Id))
+                var updatedDayTimeRangeIds =
+                    updateAvailabilityDto.Schedule.ExclusionDates.SelectMany(dto => dto.DayTimeRanges)
+                        .Select(dto => dto.Id);
+                var deletedExclusionDates = await _context.ExclusionDates.Where(date =>
+                        date.ScheduleId == scheduleId &&
+                        updatedExclusionDateIds.All(updatedDateId => updatedDateId != date.Id))
+                    .ToListAsync();
+                var deletedDayTimeRanges = await _context.DayTimeRanges
+                    .Where(range =>
+                        updatedExclusionDateIds.Any(updatedExclusionDateId =>
+                            updatedExclusionDateId == range.ExclusionDateId) && 
+                            updatedDayTimeRangeIds.All(updatedDayTimeRangeId => updatedDayTimeRangeId != range.Id))
```

Comment:
you can move this as a predicate to the delete method
Without data loading

Reply:
or add cascade deleting

Reply:
Cascade deleting is already added, but when merging daytime ranges, some ranges can be deleted(when a larger range from both sides is added to a smaller one) and this is for cases like this. I will move this as predicate